scipy/integrate/quadpack.py

```diff
@@ -363,8 +363,40 @@ def quad(func, a, b, args=(), full_output=0, epsabs=1.49e-8, epsrel=1.49e-8,
         else:
             warnings.warn(msg, IntegrationWarning)
             return retval[:-1]
-    else:
-        raise ValueError(msg)
+
+    elif ier == 6:  # Forensic decision tree when QUADPACK throws ier=6
+        if epsabs <= 0:  # Small error tolerance - applies to all methods
+            if epsrel <= 50 * sys.float_info.epsilon or epsrel < 5e-29:
+                msg =- "If ``errabs<=0``, `epsrel` must be greater than both 5e-29 and 50*(machine epsilon)."
+            elif weight in ['sin', 'cos'] and (abs(a) + abs(b) == Inf):
+                msg = "sin- or cos- weighted intergals with infinite bounds must have ``epsabs>0``."
+
+        elif weight is None:
+            if points is None:  # QAGSE/QAGIE
+                msg = "Number of subintervals `limit` must be >=1."
+            else:  # QAGPE
+                if max(points) > max(a,b) or min(points) < min(a,b):
+                    msg = "All break points in `points` must lie within the integration bonds."
+                elif len(points) >= limit:
+                    msg = "Number of break points ({:d}) must be less than subinterval limit ({:d})".format(len(points), limit)
+
+        else:
+            if maxp1 < 1:
+                msg = "Chebyshev moment limit `maxp1` must be >=1."
+
+            elif weight in ('cos', 'sin') and abs(a+b) == Inf:  # QAWFE
+                msg = "`limlst` must be >=3."
+
+            elif weight[:3] == 'alg':  # QAWSE
+                if min(wvar) < -1:
+                    msg = "`wvar` parameters ``(alpha,beta)`` must both be >= -1."
+                if b < a:
+                    msg = "Integration limits `a`,`b` must satistfy ``a<b``."
+
+            elif weight == 'cauchy' and wvar in (a, b):
+                msg = "Parameter `wvar` must not equal integration limits `a` or `b`."
+
```

Comment:
In general, the backticks are meant for markdown syntax. Error messages need single or double quotes to appear correctly on the terminal-like places hence instead of \` you need to use `'`.

Reply:
Is there an accepted convention for referring to a variable in an Exception?

Reply:
Backticks used to be a shortcut for `__repr__` method of objects and removed in Python 3. For strings you can alternate between double and single quotes or you can escape with backslash such as `" don't need to escape these ' ' but this needs escaped \" "`